fdbserver/LogSystem.cpp

```diff
@@ -272,6 +272,15 @@ void LogSet::getPushLocations(VectorRef<Tag> tags, std::vector<int>& locations,
 	//	.detail("Included", alsoServers.size()).detail("Duration", timer() - t);
 }
 
+LogPushData::LogPushData(Reference<ILogSystem> logSystem, int tlogCount) : logSystem(logSystem), subsequence(1) {
+	ASSERT(tlogCount > 0);
+	messagesWriter.reserve(tlogCount);
+	for (int i = 0; i < tlogCount; i++) {
+		messagesWriter.emplace_back(AssumeVersion(g_network->protocolVersion()));
```

Comment:
Can "AssumeVersion(g_network->protocolVersion()" be different for different entries? If not, "reserve()" and this line can be combined with "assign()" and avoid this "for" loop?

Reply:
All entries are the same. `assign()` is c++ 20 feature, not available now. Each entry is `BinaryWriter`, which is `NonCopyable`, that's why I construct them in the for loop here.

Reply:
Ok.